kernel/Kconfig.smp

```
@@ -126,4 +126,15 @@ config TICKET_SPINLOCKS
 	  which resolves such unfairness issue at the cost of slightly
 	  increased memory footprint.
 
+config CURRENT_CPU_VALIDATE
+	bool "Validate usage of _current_cpu"
+	depends on SMP && ASSERT
+	default y
+	help
+	  Use of the _current_cpu pointer cannot be done safely in a preemptible
+	  context. If a thread is preempted and migrates to another CPU, the old
+	  CPU record will be wrong. Add a validation assert to catch incorrect
+	  usages of `_current_cpu`.
+
+	  *If you do not know what is this, leave this selected.
```

I think the wording of this could be changed slightly. The general `default y` should suffice... although I do wish we had some canned phrases for "say y here, unless you really know what you're doing" without sounding mean.

```suggestion
	  The _current_cpu pointer should only ever be used in non-preemptible
	  contexts. When assertions are enabled in SMP systems, this option enables
	  extra runtime checks to validate _current_cpu usage. 
```